modules/diskq/logqueue-disk.c

```diff
@@ -191,7 +191,10 @@ static void
 _restart_diskq(LogQueueDisk *self)
 {
   gchar *filename = g_strdup(qdisk_get_filename(self->qdisk));
-  qdisk_stop(self->qdisk);
+
+  gboolean persistent;
+  if (self->save_queue)
+    self->save_queue(self, &persistent);
```

Comment:
If I understand correctly this resonates with your other requested change comment. If I implement keeping the messages in the memory queues in case of a corrupted reload, I think this should resolve.